### PR TITLE
Switching from vLLM v0.3.3 to vLLM 0.5.4

### PR DESCRIPTION
**Pull request by gc-fu**
## Description

This PR basically switches the vLLM used in ipex-llm from v0.3.3 to v0.5.4

### 1. Why the change?

Change from vLLM 0.3.3 to vLLM 0.5.4

### 2. User API changes

None

### 3. How to test?
- [ ] N/A
- [x] Unit test: Please manually trigger the PR Validation [here](https://github.com/intel-analytics/ipex-llm-workflow/actions/workflows/llm-PR-validation.yml) by inputting the PR number (e.g., `1234`). And paste your action link here once it has been successfully finished.
- [x] Accuracy test
- [ ] Application test
- [ ] Document test
- [ ] ...

### CODE REVIEW

File: .github/actions/llm/download-llm-binary/action.yml [hunk 1]
```diff
@@ -10,7 +10,7 @@ runs:
   using: "composite"
   steps:
     - name: Download all build files
-      uses: actions/download-artifact@v3
+      uses: actions/download-artifact@4.1.7
```
Comment (gc-fu) [inline .github/actions/llm/download-llm-binary/action.yml:13]: We need to revert this.

**Comment by gc-fu** [comment]
Accuracy test result: https://github.com/intel-analytics/ipex-llm-workflow/actions/runs/10787128939

**Comment by gc-fu** [comment]
PR validation test result: https://github.com/intel-analytics/ipex-llm-workflow/actions/runs/10787221097

**Review by hzjane — APPROVED** [review]
LGTM About my changes.

**Review by xiangyuT — APPROVED** [review]
LGTM

**Review by glorysdj — APPROVED** [review]
LGTM

**Pull request merged**